core/src/main/scala/org/apache/spark/deploy/history/ApplicationHistoryProvider.scala

```diff
@@ -99,6 +104,19 @@ private[history] abstract class ApplicationHistoryProvider {
   }
 
   /**
+   * Bind to the History Server: threads should be started here; exceptions may be raised
+   * Start the provider: threads should be started here; exceptions may be raised
+   * if the history provider cannot be started.
+   * The base implementation contains a re-entrancy check and should
```

Comment:
All the work on Yarn service model, and that of SmartFrog before it, have given me a fear of startup logic. I'll see about doing it there though